numpy/linalg/linalg.py

```diff
@@ -2610,12 +2610,12 @@ def norm(x, ord=None, axis=None, keepdims=False):
 
 # multi_dot
 
-def _multidot_dispatcher(arrays):
-    return arrays
+def _multidot_dispatcher(arrays, *, out=None):
+    return tuple(arrays) + (out,)
```

Comment:
If it matches concatenate,
```suggestion
    yield from arrays
    yield out
```

Reply:
Ah thanks, should have looked that one up. Although I am a bit curious whether that is not just slower in the typical use-case :). Lets use that.

Reply:
Guessing that means it _does_ match then?

Reply:
Sorry, not quite following. Are you unsure if `tuple(something)` and `yield from something` behave differently here? I think tuple accepts any iterable just like yield from?

In any case, it does not matter, this is the pattern we use everywhere, so it should be the pattern here, and I did not even consider timing it yet.

Reply:
No, I'm just unsure whether `concatenate_dispatcher` actually contains what I think it contains (`yield`s)

Reply:
Well, not sure what concatenate does. But a lot of functions use this pattern (e.g. einsum, choose). On the other hand, there is a concatenate mockup in the tests and NEP which explicitely checks for `out=None`.
That is a bit confusing, but I suppose, since `None` does not have an `__array_function__` attribute it does not matter (and we may have a fast path for it). @shoyer do you have a quick opinion about this?